grpc/server.go

```diff
@@ -56,7 +57,11 @@ func (s *grpcWrapper) Subscribe(req *SubscribeRequest, srv PubSub_SubscribeServe
 
 	ctx, cancel := context.WithCancel(srv.Context())
 	defer cancel()
-	c, e := subscriber.Start(ctx, req.GetFilter())
+
+	// Change int64 to time.Duration
+	retentionPeriod := time.Duration(req.GetRetentionPeriod()) * time.Second
+	visibilityTimeout := time.Duration(req.GetVisibilityTimeout()) * time.Second
```

Comment:
what if these fields are not provided? They'd get set to 0 on the server...

Reply:
No I have made it so that there are default values. If the functional options "retention period" or "visibility timeout" are not provided then it just uses the default values I have set in the Start method.

Reply:
except that you pass them in unconditionally... if the client didn't specify one of these fields, then the grpc server will fill it in with the default value (0), which in turn gets passed into the `Start` function here, which then gets set on line 48 in `grpc/client.go`

Run a test to verify, if you wish :)